plugins/provisioners/salt/default_downloader.rb

```diff
@@ -0,0 +1,93 @@
+require 'open-uri'
+require 'digest'
+require_relative "./errors"
+
+module VagrantPlugins
+  module Salt
+    class DefaultDownloader
+      BASE_URL = "https://repo.saltproject.io/bootstrap/stable/"
+      PS1 = "bootstrap-salt.ps1"
+      SH  = "bootstrap-salt.sh"
+      SHA256_SUFFIX = "sha256"
+      SIG_SUFFIX = "asc"
+      KEY_URL = "#{BASE_URL}SALT-PROJECT-GPG-PUBKEY-2023.gpg"
+
+      def initialize(communicator)
+        @communicator = communicator
+        @logger  = Log4r::Logger.new("vagrant::salt::default_downloader")
+      end
+
+      def source_url 
+        @communicator == :winrm ? "#{BASE_URL}#{PS1}" : "#{BASE_URL}#{SH}"
```

Comment:
Script should be chosen based on guest platform where if it is `:windows`, then `.ps1` is used, otherwise `.sh` is used.